ui/app/components/app/transaction-time-remaining/transaction-time-remaining.container.js

```diff
@@ -0,0 +1,84 @@
+import { connect } from 'react-redux'
+import {
+    getEstimatedGasPrices,
+    getEstimatedGasTimes,
+    getCustomGasPrice,
+    formatTimeEstimate,
+    getFastPriceEstimateInHexWEI,
+} from '../../../selectors/custom-gas'
+import TransactionTimeRemaining from '../transaction-time-remaining/transaction-time-remaining.component'
+import { getAdjacentGasPrices, extrapolateY } from '../gas-customization/gas-price-chart/gas-price-chart.utils'
+import { hexWEIToDecGWEI } from '../../../helpers/utils/conversions.util'
+import { getSelectedToken } from '../../../selectors/selectors.js'
+import { fetchBasicGasAndTimeEstimates } from '../../../ducks/gas/gas.duck'
+
+const mapStateToProps = (state, ownProps) => {
+    const { selectedAddressTxList } = state.metamask
+    const { modalState: { props: modalProps } = {} } = state.appState.modal || {}
+    const { transaction = {} } = ownProps
+    const { txData = {} } = modalProps || {}
+    const selectedTransaction = selectedAddressTxList.find(({ id }) => id === (transaction.id || txData.id))
+    const { gasPrice: currentGasPrice } = getTxParams(state, selectedTransaction)
+    const customModalGasPriceInHex = getCustomGasPrice(state) || currentGasPrice
+    const customGasPrice = calcCustomGasPrice(customModalGasPriceInHex)
+    const gasPrices = getEstimatedGasPrices(state)
+    const estimatedTimes = getEstimatedGasTimes(state)
+
+    return {
+        customGasPrice,
+        currentTimeEstimate: getRenderableTimeEstimate(customGasPrice, gasPrices, estimatedTimes),
+    }
+}
+
+const mapDispatchToProps = dispatch => {
+
+    return {
+        fetchBasicGasAndTimeEstimates: () => dispatch(fetchBasicGasAndTimeEstimates()),
+    }
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(TransactionTimeRemaining)
+
+function getRenderableTimeEstimate (currentGasPrice, gasPrices, estimatedTimes) {
```

Comment:
This is fine for this PR, but we eventually to to abstract this code to some sort of utility file and then import it as needed.